### PR TITLE
Add supported platforms in README

### LINKED ISSUE #66: Document the supported compiler and platform version
Brought up by @maxgolov, something similar to this should exist in the README.md file:

Operating System	Compiler
Mac OS X 10.11.6	Clang Xcode 8.3
Mac OS X 10.12.6	Clang Xcode 9.0, 9.1
Mac OS X 10.13.3	Clang Xcode 9.2, 9.3, 10.0, 10.1
Raspbian GNU/Linux 8 (jessie)	GCC 4.9.2 (armv7l)
Ubuntu 14.04.* LTS	GCC 4.8.*, 4.9.4
Ubuntu 14.04.1 LTS	GCC 5.x.x
Ubuntu 16.04 LTS	GCC 5.x.x (armv7l)
Windows 10	Android Studio/Gradle
Windows Server 2016	Visual Studio 2017

### PR DESCRIPTION
**Pull request by lalitb**
This fixes #66

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,19 @@ Please note that supporting the [C Programming
 Language](https://en.wikipedia.org/wiki/C_(programming_language)) is not a goal
 of the current project.
 
+## Supported Platforms
```
Comment (ThomsonTan) [inline README.md:25]: Should this be `Supported Development Platforms` as we don't want to guarantee our code to just run on below platforms?

Comment (lalitb) [reply README.md:25]: Good point. Will make that explicit.

Comment (lalitb) [reply README.md]: done.

**Review by ThomsonTan — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -22,6 +22,19 @@ Please note that supporting the [C Programming
 Language](https://en.wikipedia.org/wiki/C_(programming_language)) is not a goal
 of the current project.
 
+## Supported Development Platforms
+
+ Our CI pipeline builds and tests on following platforms:
```
Comment (jsuereth) [inline README.md:27]: Should we also denote CPU architecture?  I think these are all x86 IIUC.

I've seen comments on ARM, Android, embedded support.

**Review by jsuereth — APPROVED** [review]
Just one minor clarification that would be nice.

File: README.md (hunk 2 above)
Comment (lalitb) [reply README.md:27]: Good point, yes these are xx6. Will modify it.

Comment (ThomsonTan) [reply README.md]: To be specific, our CI only run for x86_64, not x86 in terms of 32-bit?

Comment (lalitb) [reply README.md]: Yes correct. - I though of clarifying in terms of x86 ( Intel / PC ) vs ARM ( mobile / embedded devices ). Will make it more specific : )

**Pull request merged**